helper/bearer.go

```diff
@@ -25,9 +25,28 @@ import (
 	"strings"
 )
 
-func BearerTokenFromRequest(r *http.Request) string {
-	auth := r.Header.Get("Authorization")
-	split := strings.SplitN(auth, " ", 2)
+const (
+	defaultAuthorizationHeader = "Authorization"
+)
+
+type BearerTokenLocation struct {
+	Header         *string `json:"header"`
+	QueryParameter *string `json:"query_parameter"`
+}
+
+func BearerTokenFromRequest(r *http.Request, tokenLocation *BearerTokenLocation) string {
+	var token string
+	if tokenLocation != nil {
+		if tokenLocation.Header != nil {
+			token = r.Header.Get(*tokenLocation.Header)
```

Comment:
Wouldn't it make more sense to return the value here directly?

```suggestion
			return r.Header.Get(*tokenLocation.Header)
```

Reply:
We can't return the value directly there, as in your suggestion, because we need to return only the token without the "bearer" part. However, I thought about moving lines 49-53 to a separate function. Then the line of your suggestion would look like:
```
return getToken(r.Header.Get(*tokenLocation.Header))
```

Reply:
What do you think?

Reply:
The `bearer` is only relevant when the header is `Authorization` as as means of switching authorization modes (e.g. basic, bearer, ...).

However, an explicit query parameter such as `?token=1234` would not include the bearer instruction as the parameter is already explicitly bound to a token. The same applies if we define a *custom* header such as `X-MyApplication-AuthToken`. Here too, the context is explicit and clear.

Reply:
I think in query param schema should not be present in the value but in custom header it coud be, this gives more guarantee/flexibility that some libs in developer  application will still work. OAuth for example expects `Bearer` scheme

Reply:
We can also make that optional for Header. So both `bearer token` and `token` will work. @aeneasr how would you prefere?

Reply:
> I think in query param schema should not be present in the value but in custom header it coud be, this gives more guarantee/flexibility that some libs in developer application will still work. OAuth for example expects Bearer scheme

That would be two authenticators with two separate custom headers IMO

Reply:
@aeneasr We do not give the possibility to override the scheme and IMO it should remain as it is but with the option to use different header

Reply:
@aeneasr the last word is yours. Do you want the "bearer" in custom header value to be optional, required or not valid?

Reply:
I think it's very confusing to have the "bearer" in the custom header and the query parameter. If you look at the documentation it also does not become very clear. This is something I believe will make things very awkward when people are trying to use these features as I've never ever in my life have seen something like `X-Token-Auth: bearer bla` nor `?token=bearer%20bla`. So please remove the "bearer" splitting for these two parameters and, if unclear, explicitly document it with examples in the docs.

Reply:
Ok, done 😄